packages/protocol/contracts/governance/DowntimeSlasher.sol

```diff
@@ -35,99 +53,198 @@ contract DowntimeSlasher is SlasherUtil {
+   * @return True if the validator signature does not appear in any block within the window.
+   */
+  function wasDownForIntervals(
+    uint256[] memory startBlocks,
+    uint256[] memory endBlocks,
+    uint256[] memory signerIndices
+  ) public view returns (bool) {
+    require(startBlocks.length > 0, "requires at least one interval");
+    require(
+      startBlocks.length == endBlocks.length,
+      "startBlocks and endBlocks must have the same length"
+    );
+    require(signerIndices.length > 0, "requires at least one signerIndex");
+
+    uint256 epochSize = getEpochSize();
+    uint256 signerIndicesIndex = 0;
+    for (uint256 i = 0; i < startBlocks.length; i = i.add(1)) {
+      if (i > 0) {
+        require(
+          startBlocks[i.sub(1)] < startBlocks[i],
+          "each interval must start after the start of the previous interval"
+        );
+        require(
+          startBlocks[i] <= endBlocks[i.sub(1)].add(1),
+          "each interval must start at most one block after the end of the previous interval"
+        );
+        require(
+          endBlocks[i.sub(1)] < endBlocks[i],
+          "each interval must end after the end of the previous interval"
+        );
+        // The signer index of a particular validator may change from epoch to epoch.
+        // Because the intervals for which bitmaps are calculated in this contract do not span
+        // epochs, and because intervals processed by this function are guaranteed to be
+        // overlapping or contiguous, whenever we cross epoch boundaries we are guaranteed to
+        // process an interval that starts with the first block of that epoch.
+        if (startBlocks[i].sub(1).mod(epochSize) == 0) {
+          require(
+            getValidatorAccountFromSignerIndex(
+              signerIndices[signerIndicesIndex],
+              startBlocks[i].sub(1)
+            ) ==
+              getValidatorAccountFromSignerIndex(
+                signerIndices[signerIndicesIndex.add(1)],
+                startBlocks[i]
+              ),
+            "indices do not point to the same validator"
+          );
+          signerIndicesIndex = signerIndicesIndex.add(1);
+        }
+      }
+      if (!wasDownForInterval(startBlocks[i], endBlocks[i], signerIndices[signerIndicesIndex])) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  /**
+   * @notice Slashes a validator that did not sign any blocks for at least `slashableDowntime`.
+   * @param startBlocks A list of interval start blocks for which signature bitmaps have already
+   * been set.
+   * @param endBlocks A list of interval end blocks for which signature bitmaps have already
+   * been set.
+   * @param signerIndices The index of the provided validator for each epoch over which the
+   * provided intervals span.
    * @param groupMembershipHistoryIndex Group membership index from where
-   * the group should be found. (For start block)
+   * the group should be found (For start block).
```

Comment:
We should clarify the parameters below. It could either be clarified inline or we could link somewhere.

Reply:
Well, this is using the SlasherUtil.sol, but it doesn't have any documentation.
I could change the docs for the SlasherUtil, but will be different from the deployed one, that shouldn't be a problem but I don't know if we assuming a version change for comments (@asaj)
The other thing that I could do, is to add a link to the `slash` function in the `lockedGold`
 contract, that's basically what the SlasherUtils end up calling for the validator, and for the group. WDYT?